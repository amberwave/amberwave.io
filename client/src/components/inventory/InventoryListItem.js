import React from 'react';
import { Link } from 'react-router-dom';
import moment from 'moment';
import numeral from 'numeral';

const InventoryListItem = ({
  _id,
  name,
  model,
  location,
  history,
  connection,
  status,
}) => (
  <tr>
    <td>{name}</td>
    <td>{model}</td>
    <td>{_id}</td>
    <td>
      ({location.latitude}, {location.longitude}, {location.radius})
    </td>
    <td>{history.lastConnection}</td>
    <td>
      {connection.exchange}/{connection.queue}
    </td>
    <td>{status}</td>
  </tr>
);

export default InventoryListItem;
