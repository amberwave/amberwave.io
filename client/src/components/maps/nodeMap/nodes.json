[
  {
    "key": "New York",
    "name": "drollsRindier",
    "type": "EPA",
    "status": "running",
    "topic": "/drollsRindier/epa",
    "coordinates": {
      "latitude": 40.6643,
      "longitude": -73.9385
    }
  },
  {
    "key": "Los Angeles",
    "name": "deforestersDeerlike",
    "type": "Lite",
    "status": "running",
    "topic": "/deforestersDeerlike/lite",
    "coordinates": {
      "latitude": 34.0194,
      "longitude": -118.4108
    }
  },
  {
    "key": "Chicago",
    "name": "interdictorsDopily",
    "type": "Lite",
    "status": "suspended",
    "topic": "/interdictorsDopily/lite",
    "coordinates": {
      "latitude": 41.8376,
      "longitude": -87.6818
    }
  },
  {
    "key": "Houston",
    "name": "staphylomataJudiciary",
    "type": "heavy",
    "status": "suspended",
    "topic": "/staphylomataJudiciary/heavy",
    "coordinates": {
      "latitude": 29.7805,
      "longitude": -95.3863
    }
  },
  {
    "key": "Phoenix",
    "name": "squealersResidue",
    "type": "heavy",
    "status": "shutdown",
    "topic": "/squealersResidue/heavy",
    "coordinates": {
      "latitude": 33.5722,
      "longitude": -112.088
    }
  },
  {
    "key": "Philadelphia",
    "name": "rivalingFained",
    "type": "midspeed",
    "status": "running",
    "topic": "/rivalingFained/midspeed",
    "coordinates": {
      "latitude": 40.0094,
      "longitude": -75.1333
    }
  },
  {
    "key": "San Antonio",
    "name": "creepagesZanders",
    "type": "lite",
    "status": "shutdown",
    "topic": "/creepagesZanders/lite",
    "coordinates": {
      "latitude": 29.4724,
      "longitude": -98.5251
    }
  },
  {
    "key": "San Diego",
    "name": "slingingFullfaces",
    "type": "midspeed",
    "status": "suspended",
    "topic": "/slingingFullfaces/midspeed",
    "coordinates": {
      "latitude": 32.8153,
      "longitude": -117.135
    }
  },
  {
    "key": "Dallas",
    "name": "clocheNarratively",
    "type": "midspeed",
    "status": "running",
    "topic": "/clocheNarratively/midspeed",
    "coordinates": {
      "latitude": 32.7757,
      "longitude": -96.7967
    }
  },
  {
    "key": "San Jose",
    "name": "praisefulRizzarts",
    "type": "lite",
    "status": "running",
    "topic": "/praisefulRizzarts/lite",
    "coordinates": {
      "latitude": 37.2969,
      "longitude": -121.8193
    }
  }
]
