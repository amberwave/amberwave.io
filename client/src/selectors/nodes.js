const getVisibleNodes = (nodes, { text, sortBy }) =>
  nodes
    .filter((node) => {
      const textMatch = node.name.toLowerCase().includes(text.toLowerCase());
      return textMatch;
    })
    .sort((a, b) => {
      if (sortBy === 'date') {
        return a.date < b.date ? 1 : -1;
      }
      if (sortBy === 'type') {
        return a.type < b.type ? 1 : -1;
      }
      if (sortBy === 'status') {
        return a.status < b.status ? 1 : -1;
      }
      if (sortBy === 'dataType') {
        return a.dataType < b.dataType ? 1 : -1;
      }
    });

export default getVisibleNodes;
