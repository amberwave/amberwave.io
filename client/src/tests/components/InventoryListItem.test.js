import InventoryListItem from '../../components/inventory/InventoryListItem';
import React from 'react';
import nodes from '../fixtures/nodes';
import { shallow } from 'enzyme';

test('should render InventoryListItem with node data', () => {
  const wrapper = shallow(
    <InventoryListItem key={nodes[0]._id} {...nodes[0]} />
  );
  expect(wrapper).toMatchSnapshot();
});
