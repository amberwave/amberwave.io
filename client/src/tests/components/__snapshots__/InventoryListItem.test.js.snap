// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should render InventoryListItem with node data 1`] = `
<tr>
  <td>
    superpatriotismsProoemion
  </td>
  <td>
    EPA
  </td>
  <td>
    1
  </td>
  <td>
    (
    90
    , 
    127.12
    , 
    10
    )
  </td>
  <td>
    <Component />
  </td>
  <td>
    superpatriotismsProoemion
    /
    ambient-light
  </td>
  <td>
    suspended
  </td>
</tr>
`;
